lib/discourse_theme/cli.rb

```diff
@@ -167,6 +183,94 @@ def run(args)
+        index = dir.index("/spec")
+        spec_path = "/spec"
+
+        if index
+          spec_path = dir[index..-1]
+          dir = dir[0..index - 1]
+        end
+
+        raise DiscourseTheme::ThemeError.new "'#{dir} does not exist" unless Dir.exist?(dir)
+
+        # Checks if the container is running
+        container_name = DISCOURSE_TEST_DOCKER_CONTAINER_NAME
+        is_running = false
+        container_exists = false
+
+        if !(
+             output =
+               execute(command: "docker ps -a --format '{{json .}}' | grep #{container_name}")
+           ).empty?
+          container_exists = true
+          is_running = JSON.parse(output)["State"] == "running"
+        end
+
+        basename = Pathname.new(dir).basename.to_s
+        verbose = !!args.delete("--verbose")
+        headless = !!args.delete("--headless")
+
+        if !is_running || args.delete("--rebuild")
+          if container_exists
+            execute(command: "docker stop #{container_name}")
+            execute(command: "docker rm -f #{container_name}")
+          end
+
+          execute(
+            command: <<~CMD.squeeze(" "),
+              docker run -d \
+                -p 31337:31337 \
+                --add-host host.docker.internal:host-gateway \
+                --entrypoint=/sbin/boot \
+                --name=#{container_name} \
+                -v #{dir}:/tmp/#{basename} \
+                discourse/discourse_test:release
+            CMD
+            message: "Creating discourse/discourse_test:release Docker container...",
+            stream: verbose,
+          )
+
+          execute(
+            command:
+              "docker exec -u discourse:discourse #{container_name} ruby script/docker_test.rb --no-tests --checkout-ref origin/tests-passed",
+            message: "Checking out latest Discourse source code...",
+            stream: verbose,
+          )
+
+          execute(
+            command:
+              "docker exec -u discourse:discourse #{container_name} bundle exec rake docker:test:setup",
+            message: "Setting up Discourse test environment...",
+            stream: verbose,
+          )
+
+          execute(
+            command: "docker exec -u discourse:discourse #{container_name} bin/ember-cli --build",
+            message: "Building Ember CLI assets...",
+            stream: verbose,
+          )
+        end
+
+        rspec_envs = []
+
+        if headless
+          WebDriver.start(browser: :chrome)
+
+          rspec_envs.push("SELENIUM_HEADLESS=0")
+          rspec_envs.push("CAPYBARA_SERVER_HOST=0.0.0.0")
+          rspec_envs.push("CAPYBARA_REMOTE_DRIVER_URL=http://host.docker.internal:9515")
+        end
+
+        rspec_envs = rspec_envs.map { |env| "-e #{env}" }.join(" ")
+
+        execute(
+          command:
+            "docker exec #{rspec_envs} -u discourse:discourse #{container_name} bundle exec rspec --color --tty #{File.join("/tmp", basename, spec_path)}".squeeze(
```

Comment:
I wonder if a `-t` is missing from `docker exec` and that is why you must pass `--color` and `--tty` to rspec.

Reply:
O thanks! This was it :) I have made the change.